### DISCUSSION TITLE
Set metadata inside server-side component

### DISCUSSION THREAD

**Discussion by gquagliano** [vercel/next.js#64366] [App Router]
Hello!

I usually need the same data to render the page content and to generate metadata, for example, the page title. Therefore, with `generateMetadata`, the same data is fetched twice in every request. I understand `fetch` is cached, but I'm using Prisma. Example:

```
export async function generateMetadata({ path }) {
    const page = await getPageContent(path);
    return {
        title: page.title
    };
}

export default async function Page({ path }) {
    const page = await getPageContent(path);
    return (
        <main>
            <h1>{page.title}</h1>
            <p>{page.description}</p>
        </main>
    );
}
```

Is there a way to set metadata inside `Page`? Or do this in a way where `getPageContent` is called once? And use `headers()` inside `generateMetadata`?

If caching is to be used, keep in mind that this can be user-specific, and also host-specific (using `headers().get("host")` to determine the content to display).

Thank you.

**Comment by gquagliano** [comment]
So I just realized that if you include a `<title>` tag, it gets moved to the `<head>` section of the rendered HTML. No need to include `<head>` (it will stay where you placed it in the document structure). So that's the answer.

**Comment by zalodias** [comment]
There's an issue with including a `<title>` tag in the component. If you defined global metadata for all your pages (i.e. in `layout.tsx`), the `<title>` of the page will be added instead of replaced. So you'll have 2 `<title>` tags inside the `<head>` markup.
